src/ApertureFlux.cc

```diff
@@ -354,10 +354,15 @@ void ApertureFluxTransform::operator()(
     }
     afw::table::SourceCatalog::const_iterator inSrc = inputCatalog.begin();
     afw::table::BaseCatalog::iterator outSrc = outputCatalog.begin();
-    for (; inSrc < inputCatalog.end() && outSrc < outputCatalog.end(); ++inSrc, ++outSrc) {
-        for (std::size_t i = 0; i < _ctrl.radii.size(); ++i) {
-            FluxResult fluxResult = fluxKeys[i].get(*inSrc);
-            _magKeys[i].set(*outSrc, calib.getMagnitude(fluxResult.flux, fluxResult.fluxSigma));
+    {
+        // While noThrow is in scope, converting a negative flux to a magnitude
+        // returns NaN rather than throwing.
+        NoThrowOnNegativeFluxContext noThrow;
+        for (; inSrc < inputCatalog.end() && outSrc < outputCatalog.end(); ++inSrc, ++outSrc) {
```

Comment:
This part of these lines isn't what's been changed, but...
 - I'm not sure using `<` on iterators is safe in general; it's certainly more idiomatic to use `!=` on `for` loop exit conditions.
 - Iterator declarations seem a good place to use `auto`.